web/src/composebox_typeahead.ts

```diff
@@ -781,7 +781,13 @@ export function get_person_suggestions(
         );
 
         return suggestion_items.filter((item) =>
-            typeahead_helper.query_matches_person(query, item, should_remove_diacritics),
+            typeahead_helper.query_matches_person(
+                query,
+                item,
+                should_remove_diacritics,
+                undefined,
+                true,
```

Comment:
I don't think this correct since now the users with custom profile field matched are shown in DM typeahead as well. I think you would need to add a parameter to `get_person_suggestions` and that should be passed here instead of always passing this as true as `get_person_suggestions` is used for other typeaheads as well apart from mention typeahead.